posthog/batch_exports/sql.py

```diff
@@ -7,15 +7,23 @@
         pd.distinct_id AS distinct_id,
         toString(p.id) AS person_id,
         p.properties AS properties,
-        pd.version AS version,
-        pd._timestamp AS _inserted_at
+        pd.version AS person_distinct_id_version,
+        p.version AS person_version,
+        multiIf(
+            pd.is_updated AND NOT p.is_updated,
+            pd._timestamp,
+            p.is_updated AND NOT pd.is_updated,
+            p._timestamp,
+            least(p._timestamp, pd._timestamp)
+        ) AS _inserted_at
     FROM (
         SELECT
             team_id,
             distinct_id,
             max(version) AS version,
             argMax(person_id, person_distinct_id2.version) AS person_id,
-            max(_timestamp) AS _timestamp
+            argMax(_timestamp, person_distinct_id2.version) AS _timestamp,
+            argMax(person_distinct_id2._timestamp >= {{interval_start:DateTime64}} AND person_distinct_id2._timestamp < {{interval_end:DateTime64}}, person_distinct_id2.version) AS is_updated
```

Comment:
Sidenote: It's more often than not that clickhouse's auto resolving of aliases is an annoyance rather than a feature.